cpp/daal/src/algorithms/dtrees/forest/oneapi/cl_kernels/df_tree_level_build_helper_kernels.cl

```diff
@@ -41,33 +41,83 @@ DECLARE_SOURCE(
         treeOrder[offset + id] = treeOrderBuf[offset + id];
     }
 
-    __kernel void doLevelPartition(const __global int * data, const __global int * nodeList, const __global int * treeOrder,
-                                   __global int * treeOrderBuf, int nFeatures) {
-        const int nNodeProp  = NODE_PROPS; // num of split attributes for node
-        const int leafMark   = -1;
-        const int local_id   = get_sub_group_local_id();
-        const int nodeId     = get_global_id(1);
-        const int local_size = get_sub_group_size();
+    __kernel void doLevelPartitionByGroups(const __global int * data, const __global int * nodeList, __global int * nodeAuxList,
+                                           const __global int * treeOrder, __global int * treeOrderBuf, int nNodes, int nFeatures) {
+        const int nNodeProp    = NODE_PROPS;     // num of split attributes for node
+        const int nAuxNodeProp = AUX_NODE_PROPS; // num of auxilliary attributes for node
+        const int maxBlocksNum = PARTITION_MAX_BLOCKS_NUM;
+        const int minBlockSize = PARTITION_MIN_BLOCK_SIZE;
+        const int leafMark     = -1;
+
+        const int sub_group_size               = get_sub_group_size();
```

Comment:
There are two functions: uint    get_sub_group_size( void ); uint    get_max_sub_group_size( void );
First one is the size of current subgroup I guess so it might be better to use the second one

Reply:
thank you, will try it